src/instructlab/model/linux_test.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+from typing import Any
+import logging
+import os
+
+# Third Party
+from datasets import load_dataset
+from openai import OpenAI, Stream
+from tqdm import tqdm
+import click
+
+# First Party
+from instructlab.model.backends.llama import ensure_server
+
+# Local
+from .. import configuration as cfg
+from ..utils import get_sysprompt
+
+
+def linux_test(
+    ctx: click.Context,
+    test_file: Path,
+    models: list = [cfg.DEFAULT_MODEL],
+    create_params: dict = {},
+) -> typing.Dict[str, typing.Any]:
+    def response(user, create_params):
+        # https://platform.openai.com/docs/api-reference/chat/create
+        resp = client.chat.completions.create(
+            **create_params,
+            messages=[
+                {"role": "system", "content": get_sysprompt()},
+                {"role": "user", "content": user},
+            ],
+        )
+        if isinstance(resp, Stream):
+            # https://platform.openai.com/docs/api-reference/chat/streaming
+            response = ""
+            for r in resp:
+                if c := r.choices[0].delta.content:
+                    print(c, end="", flush=True)
+                    response += c
+            print()
+            return response
+        else:
+            # https://platform.openai.com/docs/api-reference/chat/object
+            c = resp.choices[0].message.content
+            print(c)
+            return c
+
+    ctx.obj.logger.debug("")
+
+    # https://platform.openai.com/docs/api-reference/chat/create
+    create_params["temperature"] = 0  # for more reproducible results
+    # create_params["stream"] = True  # optional, useful for interactive debugging
+
+    ds = load_dataset("json", data_files=os.fspath(test_file), split="train")
+    res: dict[str, Any] = {}
+    # intentionally not using collections.defaultdict to avoid dependency
+    for d in ds:
+        res[d["user"]] = {}
+    for m in models:
+        m = str(m)
+        ctx.obj.logger.debug("%s", m)
+        create_params["model"] = m  # mandatory but not used
+        server_process = server_queue = None
+        try:
+            server_process, endpoint_url, server_queue = ensure_server(
+                ctx.obj.logger,
+                ctx.obj.config.serve,
+                True,
+                "",
+                "",
+                "",
+                ctx.params["model_family"],
+                model_path=m,
+            )
+            ctx.obj.logger.debug("%s", endpoint_url)
+            client = OpenAI(
+                base_url=endpoint_url,
+                api_key=ctx.params["api_key"],
+            )
+            for d in ds:
+                # use markdown format
+                print()
+                print("###", d["user"])
+                print()
+                print("assistant:", d["assistant"])
+                print()
+                print(f"{m}: ", end="", flush=True)
+                res[d["user"]][m] = response(d["user"], create_params)
```

Comment:
Where's response from? Seems missing openai chat completion api call?

Reply:
`client.chat.completions.create` is within the function `response`. Would you like to suggest more suitable name?